### PR TITLE
Bump to kernel 6.6 with latest buildroot

### PR DESCRIPTION
**Pull request by jerryz123**
The 13.2 toolchain installs Linux 6.6 headers.
This updates the kernel to 6.6, and buildroot to latest, which supports 6.6.

This also provides a better error message when running marshal in a unsupported filesystem and exposes the `mount-dir` as a configurable option to resolve such errors.

### CODE REVIEW

File: boards/default/distros/br/br.py [hunk 1]
```diff
@@ -188,6 +188,22 @@ def configure(self, env):
         mergeScript = br_dir / 'merge_config.sh'
         wlutil.run([mergeScript] + kFrags, cwd=(br_dir / 'buildroot'), env=env)
 
+        # Fakeroot needs to be set to 1.31 to compile with sysroot 2.17
```
Comment (abejgonzalez) [inline boards/default/distros/br/br.py:191]: So this needs to be updated everytime the sysroot changes?

**Review by abejgonzalez — APPROVED** [review]
LGTM. Outside of Q's

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -131,6 +131,8 @@ jobs:
     steps:
       - name: Run tests (- spike tests)
         run: |
+          env
+          unset LD_LIBRARY_PATH
```
[the remark this answers is not in the document]
Comment (jerryz123) [reply .github/workflows/run-tests.yml:135]: The runner env has a bad LD_LIBRARY_PATH, its like `/ecad/xilinx/lib:` (the extra colon at the end breaks it).

File: boards/chipyard/base-workloads/br-base/buildroot-config [hunk 3]
```diff
@@ -90,7 +90,7 @@ BR2_PACKAGE_PROVIDES_TOOLCHAIN_EXTERNAL="toolchain-external-custom"
 BR2_TOOLCHAIN_EXTERNAL_PREFIX="$(ARCH)-unknown-linux-gnu"
 BR2_TOOLCHAIN_EXTERNAL_BOOTLIN_ARCH_SUPPORTS=y
 BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="$(ARCH)-unknown-linux-gnu"
-BR2_TOOLCHAIN_EXTERNAL_GCC_12=y
+# BR2_TOOLCHAIN_EXTERNAL_GCC_12 is not set
```
[the remark this answers is not in the document]
Comment (jerryz123) [reply boards/chipyard/base-workloads/br-base/buildroot-config:93]: Marshal generates its own BR2 Config fragment that has the proper GCC/linux versions set. These settings, if provided, overrides the marshal settings, potentially breaking the flow if the mismatch is catastrophic.

File: boards/chipyard/base-workloads/br-base/buildroot-config [hunk 4]
```diff
@@ -178,8 +178,8 @@ BR2_TOOLCHAIN_EXTERNAL_HAS_SSP_STRONG=y
 # BR2_TOOLCHAIN_EXTERNAL_INET_RPC is not set
 BR2_TOOLCHAIN_EXTERNAL_CXX=y
 # BR2_TOOLCHAIN_EXTERNAL_DLANG is not set
-# BR2_TOOLCHAIN_EXTERNAL_FORTRAN is not set
-# BR2_TOOLCHAIN_EXTERNAL_OPENMP is not set
+BR2_TOOLCHAIN_EXTERNAL_FORTRAN=y
+BR2_TOOLCHAIN_EXTERNAL_OPENMP=y
```
[the remark this answers is not in the document]
Comment (jerryz123) [reply boards/chipyard/base-workloads/br-base/buildroot-config:182]: Our toolchains have always provided these features. However, the newer buildroot seems to complain if it detects the feature support in the toolchain, but the config does not indicate so. This seems to be a new thing.

File: boards/default/distros/br/br.py (hunk 1 above)
Comment (jerryz123) [reply boards/default/distros/br/br.py:191]: The bug is that Conda's sysroot 2.17 contains a header file against which the host fakeroot utility at BR's version (1.34) does not compile. This workaround force downgrade's buildroot's fakeroot to an older version that does compile with 2.17. AFAICT there is no regression from this, the update to 1.34 in buildroot seems to be just routine dependency maintenance.

If conda provided a sysroot newer than 2.17 (it currently does not) and we bumped to that, then this entire override thing can be removed.

File: .github/workflows/run-tests.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (abejgonzalez) [reply .github/workflows/run-tests.yml:135]: I think this is an issue w/ the Xilinx scripts that are added to the runners `.bashrc`. I can fix this later, for now we can keep this in.

**Pull request merged**